internal/k8s/pod/pod.go

```diff
@@ -110,7 +111,8 @@ func (r *reconciler) Reconcile(ctx context.Context, _ reconcile.Request) (res re
 		pods       = make(map[string][]Pod, len(ps.Items))
 	)
 
-	for _, pod := range ps.Items {
+	for i := range ps.Items {
+		pod := ps.Items[i]
```

Comment:
Can you please describe this change?

Reply:
To avoid copy to fix the deepsource error.

Reply:
I think Range Over Copy value is less of a problem since index access to slice causes BoundaryOverflow problems.
How do you think?

Reply:
I don't mind reverting this. In that case, I think we should turn off the deepsource error by default.

Reply:
yes exactly can you add skipcq annotation for supress deepsource?

Reply:
I was thinking of turning it off globally but deepsource does not offer the feature...

[2cc4c84](https://github.com/vdaas/vald/pull/2428/commits/2cc4c8483dd99c617758454d7d2f6b3848e60543)

Reply:
thank you for revert and add annotations.